cmd/loop/liquidity.go

```diff
@@ -0,0 +1,233 @@
+	)
+	if err != nil {
+		return err
+	}
+
+	// If our config was not set before this, we set it to a non-nil value
+	// now.
+	if cfg == nil {
+		cfg = &looprpc.LiquidityConfig{}
+	}
+
+	if ctx.IsSet("inclprivate") {
+		cfg.IncludePrivate = ctx.Bool("inclprivate")
+	}
+
+	// Create a request to update our config.
+	req := &looprpc.SetLiquidityConfigRequest{
+		Config: cfg,
+	}
+
+	cfg, err = client.SetLiquidityConfig(context.Background(), req)
+	if err != nil {
+		return err
+	}
+
+	printJSON(cfg)
+
+	return nil
+}
+
+var setLiquidityRuleCommand = cli.Command{
+	Name:  "setrule",
+	Usage: "set liquidity manger rule for a target",
+	Description: "Updates the liquidity rule that we set for a target. " +
+		"At present rules can be set for your node as a whole, or on " +
+		"a per-peer level.",
+	ArgsUsage: fmt.Sprintf("[%v|%v|%v|%v]", liquidity.TargetNone,
+		liquidity.TargetNode, liquidity.TargetPeer,
+		liquidity.TargetChannel),
+	Flags: []cli.Flag{
+		cli.IntFlag{
+			Name: "mininbound",
+			Usage: "the minimum percentage of inbound liquidity " +
+				"to total capacity beneath which to " +
+				"recommend loop out to acquire inbound.",
+		},
+		cli.IntFlag{
+			Name: "minoutbound",
+			Usage: "the minimum percentage of outbound liquidity " +
+				"to total capacity beneath which to " +
+				"recommend loop in to acquire outbound.",
+		},
+	},
+	Action: setRule,
+}
+
+func setRule(ctx *cli.Context) error {
+	// We require that a target is set for this rule update.
+	if ctx.NArg() != 1 {
+		return fmt.Errorf("please set a target for the rule " +
+			"update")
+	}
+
+	client, cleanup, err := getClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	// We need to set all values in the config that we send the server to
+	// update us to. To allow users to set only individual fields on the
+	// cli, we lookup our current config, then update its values.
+	cfg, err := client.GetLiquidityConfig(
+		context.Background(), &looprpc.GetLiquidityConfigRequest{},
+	)
+	if err != nil {
+		return err
+	}
+
+	target := ctx.Args().First()
+	switch target {
+	case liquidity.TargetNone.String():
+		cfg.Target = looprpc.LiquidityTarget_NONE
+
+	case liquidity.TargetNode.String():
+		cfg.Target = looprpc.LiquidityTarget_NODE
+
+	case liquidity.TargetPeer.String():
+		cfg.Target = looprpc.LiquidityTarget_PEER
+
+	case liquidity.TargetChannel.String():
+		cfg.Target = looprpc.LiquidityTarget_CHANNEL
+
+	default:
+		return fmt.Errorf("unknown rule target: %v", target)
+	}
+
+	// Create a new rule which will be used to overwrite our current rule.
+	newRule := &looprpc.LiquidityRule{}
+
+	if ctx.IsSet("mininbound") {
+		newRule.MinimumInbound = uint32(ctx.Int("mininbound"))
+		newRule.Type = looprpc.LiquidityRuleType_THRESHOLD
+	}
+
+	if ctx.IsSet("minoutbound") {
+		newRule.MinimumOutbound = uint32(ctx.Int("minoutbound"))
+		newRule.Type = looprpc.LiquidityRuleType_THRESHOLD
+	}
+
+	ruleSet := newRule.Type != looprpc.LiquidityRuleType_UNKNOWN
```

Comment:
To reserve the default 0 value as a state that we do not recognize. Without an unknown state, we assume a type when it could just be that the caller did not set it.

Reply:
In `lnd`, most of the enums don't have this field. And for some that have it, it means that it is actually unknown. In the case of these config settings, there is no real reason to have an unknown value.

Choosing between raising questions to rpc users about the meaning of 'unknown' vs reminding users about setting a field, I'd prefer the former.